src/lib/ptree/alarm_codec.lua

```diff
@@ -288,12 +313,12 @@ function selftest ()
    end
    local function test_raise_alarm ()
       local key = {resource='res1', alarm_type_id='type1', alarm_type_qualifier=''}
-      local args = {perceived_severity='critical'}
+      local args = {perceived_severity='critical', alt_resource={'res2'}}
 
       local resource, alarm_type_id, alarm_type_qualifier = alarm:normalize_key(key)
-      local perceived_severity, alarm_text = alarm:normalize_args(args)
+      local perceived_severity, alarm_text, alt_resource = alarm:normalize_args(args)
       local alarm = {resource, alarm_type_id, alarm_type_qualifier,
-                     perceived_severity, alarm_text}
+                     perceived_severity, alarm_text, alt_resource}
```

Comment:
All this code seems OK to me. However, after reviewing the changes in `lib/yang/alarms.lua` I'm not sure it's working as expected. After this change, without any change in `lib.yang.alarms`, you should see that your `phy-down` alarm has an `alt-resource` list of alarms when the alarms is raised. You can query the list of raised alarms using `sudo ./snabb alarms get-state /'. I also noticed that the declaration of your ' phy-down` alarm doesn't have an `alt-resource` attribute.